precompiles/randomness/src/lib.rs

```diff
@@ -0,0 +1,438 @@
+pub struct RandomnessWrapper<Runtime>(PhantomData<Runtime>);
+
+impl<Runtime> Precompile for RandomnessWrapper<Runtime>
+where
+	Runtime: pallet_randomness::Config + pallet_evm::Config + pallet_base_fee::Config,
+	<Runtime as frame_system::Config>::BlockNumber: From<u32>,
+	BalanceOf<Runtime>: TryFrom<U256> + Into<U256>,
+{
+	fn execute(handle: &mut impl PrecompileHandle) -> EvmResult<PrecompileOutput> {
+		log::trace!(target: "randomness-precompile", "In randomness wrapper");
+
+		let selector = handle.read_selector()?;
+
+		// No funds are transferred to the precompile address.
+		// Transfers will directly be made on behalf of the user by the precompile.
+		handle.check_function_modifier(FunctionModifier::NonPayable)?;
+
+		match selector {
+			Action::RelayBlockNumber => Self::relay_block_number(handle),
+			Action::RelayEpochIndex => Self::relay_epoch_index(handle),
+			Action::RequestBabeRandomnessCurrentBlock => {
+				Self::request_babe_randomness_current_block(handle)
+			}
+			Action::RequestBabeRandomnessOneEpochAgo => {
+				Self::request_babe_randomness_one_epoch_ago(handle)
+			}
+			Action::RequestBabeRandomnessTwoEpochsAgo => {
+				Self::request_babe_randomness_two_epochs_ago(handle)
+			}
+			Action::RequestLocalRandomness => Self::request_local_randomness(handle),
+			Action::FulfillRequest => Self::fulfill_request(handle),
+			Action::IncreaseRequestFee => Self::increase_request_fee(handle),
+			Action::ExecuteRequestExpiration => Self::execute_request_expiration(handle),
+		}
+	}
+}
+
+impl<Runtime> RandomnessWrapper<Runtime>
+where
+	Runtime: pallet_randomness::Config + pallet_evm::Config + pallet_base_fee::Config,
+	<Runtime as frame_system::Config>::BlockNumber: TryInto<u64> + TryFrom<u64>,
+	BalanceOf<Runtime>: TryFrom<U256> + Into<U256>,
+{
+	fn relay_block_number(handle: &mut impl PrecompileHandle) -> EvmResult<PrecompileOutput> {
+		handle.record_cost(RuntimeHelper::<Runtime>::db_read_gas_cost())?;
+		let relay_block_number: u64 =
+			<Runtime as pallet_randomness::Config>::BabeDataGetter::get_relay_block_number()
+				.try_into()
+				.map_err(|_| revert("storage value is too large for provided block number type"))?;
+		Ok(succeed(
+			EvmDataWriter::new().write(relay_block_number).build(),
+		))
+	}
+	fn relay_epoch_index(handle: &mut impl PrecompileHandle) -> EvmResult<PrecompileOutput> {
+		handle.record_cost(RuntimeHelper::<Runtime>::db_read_gas_cost())?;
+		let relay_epoch_index =
+			<Runtime as pallet_randomness::Config>::BabeDataGetter::get_relay_epoch_index();
+		Ok(succeed(
+			EvmDataWriter::new().write(relay_epoch_index).build(),
+		))
+	}
+	/// Make request for babe randomness current block
+	fn request_babe_randomness_current_block(
+		handle: &mut impl PrecompileHandle,
+	) -> EvmResult<PrecompileOutput> {
+		let mut input = handle.read_input()?;
+		let contract_address = handle.context().caller;
+		let refund_address = input.read::<Address>()?.0;
+		let fee: BalanceOf<Runtime> = input
```

Comment:
@nanocryk didnt you move this check to utils? I think you can directly return the error now

Reply:
Yes all integer types are now parsed as U256 first and will error if they don't fit into the wanted type.